### PR TITLE
Treap trees

### PR DESCRIPTION
**Pull request by vgvoleg**
Signed-off-by: Oleg Ovcharuk <vgvoleg@gmail.com>

### CODE REVIEW

File: src/treap/Treap.hpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        // TODO: delete tree
+    }
+
+    void insert(Key key, T value);
+    void erase(Key key);
+
+    T* find(Key key);
+
+private:
+    TreapNode<Key, T>* root;
```
Comment (Lenferd) [inline src/treap/Treap.hpp]: = nullptr?

File: src/treap/main.cpp [hunk 2]
```diff
@@ -4,8 +4,18 @@
 #include <algorithm>
 #include <utility>
 
-#include "treap.h"
+#include "Treap.hpp"
 
 int main() {
-    return 0;
+    Treap<int, int> tree{};
```
Comment (Lenferd) [inline src/treap/main.cpp:10]: `{}` ?

**Review by Lenferd — COMMENTED** [review]
https://github.com/Lenferd/Search-trees/pull/6

File: src/treap/main.cpp (hunk 2 above)
Comment (vgvoleg) [reply src/treap/main.cpp:10]: mne eto idea podskazala

File: src/treap/Treap.hpp (hunk 1 above)
Comment (vgvoleg) [reply src/treap/Treap.hpp]: done

**Review by Lenferd — APPROVED** [review]

**Pull request merged**